lambdas/postgres-migration-async-operation/iam.tf

```diff
@@ -0,0 +1,109 @@
+data "aws_iam_policy_document" "migration_async_operation_assume_role_policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_security_group" "postgres_migration_async_operation" {
+  count = length(var.lambda_subnet_ids) == 0 ? 0 : 1
+
+  name   = "${var.prefix}-postgres-migration-async-operation"
+  vpc_id = var.vpc_id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = var.tags
+}
+
+# postgres_migration_async_operation_role
+resource "aws_iam_role" "postgres_migration_async_operation_role" {
+  name                 = "${var.prefix}-postgres_migration_async_operation"
+  assume_role_policy   = data.aws_iam_policy_document.migration_async_operation_assume_role_policy.json
+  permissions_boundary = var.permissions_boundary_arn
+  tags                 = var.tags
+}
+
+data "aws_iam_policy_document" "postgres_migration_async_operation_policy" {
+  statement {
+    actions = [
+      "ecs:RunTask",
+      "ec2:CreateNetworkInterface",
+      "ec2:DeleteNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+      "dynamodb:ListTables",
+      "lambda:GetFunction",
+      "lambda:GetFunctionConfiguration",
+      "lambda:invokeFunction",
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:DescribeLogStreams",
+      "logs:PutLogEvents",
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "s3:GetBucket*",
+      "s3:PutBucket*",
+      "s3:ListBucket*",
+    ]
+    resources = [for b in local.all_bucket_names : "arn:aws:s3:::${b}"]
+  }
+
+  statement {
```

Comment:
Does this lambda do any writing/updating/reading S3?    If not, we should remove the unneeded portion of this permission set.

Reply:
The buckets permissions seem to be crucial in allowing the ec2 task to run smoothly. I was seeing this when I removed it:
```
User: arn:aws:sts::[redacted]]:assumed-role/jtran-tf_ecs_cluster_instance_role .... is not authorized to perform: lambda:GetFunctionConfiguration on resource: arn:aws:lambda:us-east-1:[redacted]:function:jtran-tf-postgres-migration-async-operation
```

Reply:
*nods* Reads entirely make sense given this (instead of the API lambdas) is invoking the async task directly, I wasn't thinking about that quite right when I read this initially.  👍🏻 

I'd be a bit surprised if "s3:PutObject*" is required, we should be very careful giving that permission (ignoring the obvious inconsistency in our processing role being used in lambdas that probably should have a more restrictive permission set).

Reply:
It would also be good to explicitly limit the read permission to the bucket the async operation needs, this lambda shouldn't need access to anything other than the system bucket for reads iirc.

Reply:
🤦🏻   My bad, the lambda index on starting an async operation calls 

```

  await s3().putObject({
    Bucket: payloadBucket,
    Key: payloadKey,
    Body: JSON.stringify(payload),
  }).promise();
  ```

Reply:
clearly write perms are needed, at least to the payloadBucket, which should be the systemBucket.       We should restrict the bucket list though.

Reply:
[`b5b0a68` (#2165)](https://github.com/nasa/cumulus/pull/2165/commits/b5b0a681cb835ace49dd0ebfa89ebebceb2395f6)

Reply:
Nice work!